pkg/agent/params.go

```diff
@@ -85,22 +85,23 @@ func assignKubernetesParametersFromAgentProfile(profile *datamodel.AgentPoolProf
 		addValue(parametersMap, "runcVersion", config.RuncVersion)
 	}
 	addValue(parametersMap, "runcPackageURL", config.RuncPackageURL)
-	if profile.KubernetesConfig != nil && profile.KubernetesConfig.ContainerRuntime != "" {
-		// override containerRuntime parameter value if specified in AgentPoolProfile
-		// this allows for heteregenous clusters
-		addValue(parametersMap, "containerRuntime", profile.KubernetesConfig.ContainerRuntime)
-		if profile.KubernetesConfig.ContainerRuntime == "containerd" {
-			addValue(parametersMap, "cliTool", "ctr")
-			if config.ContainerdVersion != "" {
-				addValue(parametersMap, "containerdVersion", config.ContainerdVersion)
-			}
-			if config.TeleportdPluginURL != "" {
-				addValue(parametersMap, "teleportdPluginURL", config.TeleportdPluginURL)
-			}
-			addValue(parametersMap, "containerdPackageURL", config.ContainerdPackageURL)
-		} else {
-			addValue(parametersMap, "cliTool", "docker")
+	if profile.KubernetesConfig == nil || profile.KubernetesConfig.ContainerRuntime == "" {
```

Comment:
isn't this incorrect? empty string needs to hit the docker else on L104

Reply:
`if profile.KubernetesConfig != nil && profile.KubernetesConfig.ContainerRuntime != "" {` does not have an else block. The 104 else - `addValue(parametersMap, "cliTool", "docker")` is for the if on line 94 `if profile.KubernetesConfig.ContainerRuntime == "containerd" {`

The old if would not run if KubernetesConfig was nil or if ContainerRuntime was empty. Is that not the expected behavior? I will look into the docker unit test as described in your below comment

Reply:
yeah this change looks correct to me